File: cup/Laberinto.java
import java.util.ArrayList;

public class Laberinto {
    private String nombre;
    private Directiva directiva;
    private Integer[] dimension;
    private ArrayList<Elemento> monedas;
    private ArrayList<Elemento> enemigos;
    private ArrayList<Coordenada> localizaciones;
    private Coordenada inicio;
    private Coordenada meta;

    public Laberinto (String nombre, Integer[] dimension, ArrayList<Elemento> monedas, ArrayList<Elemento> enemigos, ArrayList<Coordenada> localizaciones,
        Coordenada inicio, Coordenada meta){
        this.nombre=nombre;
        this.localizaciones=localizaciones;
        this.dimension=dimension;
        this.monedas=monedas;
        this.enemigos=enemigos;
        this.inicio=inicio;
        this.meta=meta;
    }


    public Laberinto (Coordenada inicio, Coordenada meta){
        this.inicio=inicio;
        this.meta=meta;
    }
    public Laberinto (ArrayList<Coordenada> localizaciones,
    Coordenada inicio, Coordenada meta){
        this.localizaciones=localizaciones;
        this.inicio=inicio;
        this.meta=meta;
    }

    public String getNombre(){
        return this.nombre;
    }

    public Directiva getDirectiva(){
        return directiva;
    }

    public Integer[] getDimension(){
        return this.dimension;
    }

    public ArrayList<Elemento> getMonedas(){
        return this.monedas;
    }

    public ArrayList<Elemento> getEnemigos(){
        return this.enemigos;
    }

    public Coordenada getInicio(){
        return this.inicio;
    }
    
    public Coordenada getMeta(){
        return this.meta;
    }

    public ArrayList<Coordenada> getLocalizaciones(){
        return this.localizaciones;
    }


    
    public void setNombre(String nombre){
        this.nombre=nombre;
    }

    public void setDirectiva(Directiva directiva){
        this.directiva=directiva;
    }

    public void setDimension(Integer[] dimension){
        this.dimension=dimension;
    }

    public void setMonedas(ArrayList<Elemento> monedas){
        this.monedas=monedas;
    }

    public void setEnemigos(ArrayList<Elemento> enemigos){
        this.enemigos=enemigos;
    }

    public void setInicio(Coordenada inicio){
        this.inicio=inicio;
    }
    
    public void setMeta(Coordenada meta){
        this.meta=meta;
    }

    public void setLocalizaciones(ArrayList<Coordenada> localizaciones){
        this.localizaciones=localizaciones;
    }    



}

File: antlr_c/antlr_c/Clases/Laberinto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace antlr_c.Clases
{
    public class Laberinto
    {
        private String nombre;
        private Directiva directiva;
        private int[] dimension;
        private List<Elemento> definiciones;
        private List<Elemento> localizaciones;
        private Coordenada inicio;
        private Coordenada meta;

        
        public Laberinto() { }
        public Laberinto(String nombre, Directiva directiva, int[] dimension, List<Elemento> monedas, List<Elemento> enemigos,
            List<Elemento> localizaciones, Coordenada inicio, Coordenada meta)
        {
            this.nombre = nombre;
            this.localizaciones = localizaciones;
            this.directiva = directiva;
            this.dimension = dimension;
            //modificar
            //monedas.Add(enemigos);
            this.definiciones = monedas;
            this.inicio = inicio;
            this.meta = meta;            
        }
        public Laberinto(String nombre, Directiva directiva, int[] dimension, Coordenada inicio, Coordenada meta) {
            this.nombre = nombre;
            this.directiva = directiva;
            this.dimension = dimension;
            this.inicio = inicio;
            this.meta = meta;
        }
        public Laberinto(String nombre) 
        {
            this.nombre = nombre;
        }

        public String getNombre() {
            return this.nombre;
        }
        public Coordenada getInicio() {
            return this.inicio;
        }
        public Coordenada getMeta() {
            return this.meta;
        }

        public List<Elemento> getLocalizaciones() {
            return this.localizaciones;
        }

        public int[] getDimension() {
            return this.dimension;
        }

        public Directiva getDirectiva() {
            return this.directiva;
        }

        public void setNombre(String nombre) {
            this.nombre = nombre;
        }

        public void setDirectiva(Directiva directiva) {
            this.directiva = directiva;
        }
        public void setDimension(int[] dimension)
        {
            this.dimension = dimension;
        }

        public void setDefinicion(List<Elemento> definiciones)
        {
            this.definiciones = definiciones;
        }


        public void setInicio(Coordenada inicio)
        {
            this.inicio = inicio;
        }

        public void setMeta(Coordenada meta)
        {
            this.meta = meta;
        }

        public void setLocalizaciones(List<Elemento> localizaciones)
        {
            this.localizaciones = localizaciones;
        }

        public String definicionToString()
        {
            String str = "";
            for (int i = 0; i < this.definiciones.Count; i++)
            {
                str = str + "\t<definicion>\n\t" + this.definiciones.ElementAt(i) + "</definicion>\n";
            }
            return str;
        }
    }
}
